cmd/pw_ws_broker/web.go

```diff
@@ -175,16 +199,25 @@ func (bw *PwWsBrokerWeb) indexPage(w http.ResponseWriter, r *http.Request) {
 }
 
 func (bw *PwWsBrokerWeb) jsonGrid(w http.ResponseWriter, r *http.Request) {
-	grid := tile_grid.GetGrid()
+	if r.Header.Get("If-None-Match") == gridJsonPayloadETag {
+		w.WriteHeader(304)
+		return
+	}
 	w.Header().Set("Content-Type", "application/json")
 	w.Header().Set("Cross-Origin-Resource-Policy", "cross-origin")
 	w.Header().Set("Access-Control-Allow-Origin", "*")
-	json := jsoniter.ConfigFastest
-	buf, err := json.MarshalIndent(grid, "", "  ")
-	if nil != err {
-		w.WriteHeader(500)
+	w.Header().Set("ETag", gridJsonPayloadETag)
+	w.Header().Set("Cache-Control", "public, max-age=86400")
```

Comment:
We should really have a caching / nginx layer in front that controls this eventually but okay for now.